nengo_gui/static/top_toolbar.js

```diff
@@ -49,6 +50,19 @@ Nengo.Toolbar = function(filename) {
         Nengo.hotkeys.callMenu();
     });
 
+    $('#Download_button')[0].addEventListener('click', function () {
+        data_items = Nengo.Component.components;
```

Comment:
This should be ```var data_items```, or don't create this variable at all, and just do ```data_to_csv(Nengo.Component.components)``` on the next line.

(leaving out the ```var``` tells javascript to make this a new global variable, which I don't think you want).